test/integration/consul-container/test/gateways/gateway_endpoint_test.go

```diff
@@ -248,3 +248,36 @@ func checkRoute(t *testing.T, ip string, port int, path string, headers map[stri
 
 	})
 }
+
+func checkRouteError(t *testing.T, ip string, port int, path string, headers map[string]string, expected string) {
+	failer := func() *retry.Timer {
+		return &retry.Timer{Timeout: time.Second * 60, Wait: time.Second * 60}
+	}
+
+	client := cleanhttp.DefaultClient()
+	url := fmt.Sprintf("http://%s:%d", ip, port)
+
+	if path != "" {
+		url += "/" + path
+	}
+
+	retry.RunWith(failer(), t, func(r *retry.R) {
+		t.Logf("making call to %s", url)
+		req, err := http.NewRequest("GET", url, nil)
+		assert.NoError(t, err)
+
+		for k, v := range headers {
+			req.Header.Set(k, v)
+
+			if k == "Host" {
+				req.Host = v
+			}
+		}
+		_, err = client.Do(req)
+		assert.Error(t, err)
+
+		if expected != "" {
+			assert.ErrorContains(t, err, expected)
+		}
```

Comment:
both expected errors are empty in the functions that call this, I'm assuming that's because the error value is changing depending on how quickly the connection is terminating or something?

Reply:
Yeaaaa, I think keeping this capability in the helper function makes sense, but I had to move away from checking for EOF due to differences in CI in https://github.com/hashicorp/consul/pull/16408/commits/a596dbf175d7446606b188c1531f3b8801770449

Reply:
That's fine then, I think so long as we keep the positive checks first (i.e. the gateway route is routable), then we should be fine here, mainly just don't want this to be a false positive because of something like the gateway not having been configured yet.